goal-planning/goal-planning-frontend
# Feat / Login & Logout

## Changes
1. Add Express routes for `login` and `logout` logic.
2. Created Login form.
3. Implement Redux thunk logic to handle login/logout states.
4. Clean up some code formatting with Prettier.
5. Fix package.json dependency versioning on `react-router-dom` to be static to fix versioning compatibility issues with node.

## Purpose
Establish a login/logout authentication feature for the Goal Planning app.

## Approach
Create the express routes to exploit the server side API request. Then use the frontend application to ping the express server to talk to our Django application and get corresponding json data and access tokens to utilize as cookies or data directly.

## Testing Steps
1. Pull down code.
2. Run your Django Server code
3. Run your Express Server code
4. Run your React Frontend code
5. Open the frontend code at `http://localhost:3000`
6. Test the login. (May need to register a user at `http://localhost:3000/register`)
7. View Application cookies on the browser inspector and verify we have an access token and a refresh token.
8. Use the React extension debugger tool and verify that our auth state is populated correctly with the correct data.
9. Test the logout button on the nav bar.
10. View the Application cookies again to verify that they are removed.
11. View the React auth states and verify that the states are reverted back to default.

Closes #54 #55 #58
